src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQueryInputFormatProvider.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.source;
+
+import io.cdap.cdap.api.data.batch.InputFormatProvider;
+import org.apache.hadoop.conf.Configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Input format provider for BigQuery source.
+ */
+public class BigQueryInputFormatProvider implements InputFormatProvider {
+
+  protected Configuration configuration;
+
+  public BigQueryInputFormatProvider(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  @Override
+  public String getInputFormatClassName() {
+    return PartitionedBigQueryInputFormat.class.getName();
+  }
+
+  @Override
+  public Map<String, String> getInputFormatConfiguration() {
+    Map<String, String> config = new HashMap<>();
```

Comment:
Can be done as
`StreamSupport.stream(configuration.spliterator(), false).collect(Collectors.toMap(Entry::getKey, Entry::getValue))`
Wondering why do we need to copy the whole configuration. In any case, it would be great to move to constructor.

Reply:
I am wondering if we should switch from making Configuration that would have all the defaults, then adding few items and changing it to a map that potentially clashes with the different defaults on the other side.

Instead we should make a map and then create a Configuration only when needed (when calling org.apache.hadoop.fs.Path#getFileSystem)

Reply:
You can use `SourceInputFormatProvider` to just return the non-default configs, there is logic to compute the non-default here - https://github.com/cdapio/hydrator-plugins/blob/develop/hydrator-common/src/main/java/io/cdap/plugin/common/SourceInputFormatProvider.java#L40. It is in hydrator-common, so you should be able to directly use it here.